feat(onboarding): service methods and task manager refactor

constantinius opened the pull request:
Closes https://linear.app/getsentry/issue/TET-396/make-backend-service-the-only-way-to-interact-with-onboarding-models
Closes https://linear.app/getsentry/issue/TET-390/refactor-manager-api-usage

This PR adds the service methods: complete_onboarding_task, has_completed_onboarding_task, transfer_onboarding_tasks.

The `sentry.receivers.onboarding` module now does not rely on the `OrganizationOnboardingTask` anymore but on the service methods.

All task closing is now done via `complete_onboarding_task`, which underneath uses `OrganizationOnboardingTaskManager.record`, thus caching results.

constantinius committed:
ref(onboarding): extracting common functionality to conditionally set project flags update the model and send a signal into set_project_flag_and_signal function. Update callsites to use that function where appropriate.

constantinius committed:
fix(onboarding): put set_project_flag_and_signal into a separate file to avoid circular imports

constantinius committed:
fix(onboarding): import fixes

constantinius committed:
test(onboarding): fix tests for new signal handling

constantinius committed:
fix(onboarding): moving from directly calling signal handler to invoking it

constantinius committed:
Merge branch 'master' into constantinius/ref/set-project-flag

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes

constantinius committed:
test(onboarding): add tests for 'set_project_flag_and_signal'

constantinius committed:
Merge branch 'master' into constantinius/ref/set-project-flag

constantinius committed:
fix(onboarding): typing issues

constantinius committed:
cleanup

constantinius committed:
feat(onboarding): add service methods to complete an onboarding task and to check whether one has already been completed or skipped

constantinius committed:
Merge branch 'constantinius/ref/set-project-flag' into constantinius/feat/onboarding/onboarding-service-methods

constantinius committed:
feat(onboarding): migrating to service functions

constantinius committed:
fix(onboarding): wrong flag name

constantinius committed:
test(onboarding): improving test coverage

constantinius committed:
Merge branch 'constantinius/ref/set-project-flag' into constantinius/feat/onboarding/onboarding-service-methods

constantinius committed:
feat(onboarding): add service method to migrate onboarding tasks from one org to another

constantinius committed:
ref(onboarding): using service methods to mark onboarding tasks as completed instead of direct model access

constantinius committed:
Merge branch 'master' into constantinius/feat/onboarding/onboarding-service-methods

constantinius committed:
fix(onboarding): passing status and wrong arguments. Add service methods to be exposed

constantinius committed:
fix(onboarding) missing self parameter

constantinius committed:
fix(onboarding) mypy issues

vgrozdanic commented on src/sentry/models/organizationonboardingtask.py:
```diff
@@ -54,21 +54,29 @@ def values(cls) -> list[int]:
             if created:
-                return True
-        return False
+                return task_model
+        return None
```
This is a bit hard to understand return pattern, why don't we always return tuple (`task_object`, `created`)?

Then there is no need to read the source code to understand when is `None` returned, and when the object is returned, and there won't be a need to refactor the code if we also want to do something with object after it was updated, because currently we can only do something with it if it was created.

constantinius committed:
ref(onboarding): complete_onboarding_task now returns a boolean
Signifying whether the task was completed during the call. Streamlining all callsites where the result was used

constantinius committed:
fix(onboarding): removed TODO and improved error message of OrganizationOnboardingTaskManager.record

constantinius committed:
Merge branch 'master' into constantinius/feat/onboarding/onboarding-service-methods
